### ISSUE TITLE
[Accessibility] Add ability to specify alternative text for avatar and profile images

### ISSUE THREAD

**Issue by jt-nti** [mastodon/mastodon#20404]
### Pitch

Alternative text can be added to images in toots, however it does not appear to be possible to specify alternative text for a user's avatar or profile image.

### Motivation

The avatar and profile images are an important part of a user's profile however it is not accessible to anyone who relies on alternative text for images.

**Comment by jt-nti** [comment]
I found a great example of someone using their profile bio to include alt text for the header image and their avatar...

![image](https://user-images.githubusercontent.com/3535067/203863604-656a1eda-1d6c-47fa-b01c-d58a826bee19.png)

Alt text for the avatar would also be useful but it would only need to be used in some contexts, such as viewing a profile, and not others, i.e. alt text for every avatar image in a thread would be bad.

https://mastodon.me.uk/@jtonline/109400714018741009

**Comment by Cheeseness** [comment]
Can I suggest including "avatar" or "profile image" in the issue title? It took me a while to find this issue when searching to see if anybody had already made one.

**Comment by b-xb** [comment]
**Regarding design and planning**

For the sake of discussion around this issue.

I wanted to link to this related issue another user added.

https://github.com/mastodon/mastodon/issues/21858

If alt text for avatars did exist, which views should it be added to?

As the other user says, it probably does makes sense, in the timeline view for example, for the avatar to be ignored completely, unless a user specifically wants to read the image description.

So in implementing the alt text, perhaps it should only be added on specific views?

In addition, if we are discussing adding alt text to the avatar here, it is probably also worth discussing adding a header image too.

**Regarding implementation**

There are a couple of ways this could be implemented.

1/ Adding an extra field to the account model
2/ Pulling the information from one of the use custom fields

The first of these would likely take longer to implement as it would have a much more significant knock on effect.

The second of these would be far easier to implement, but would be a much less clean solution for the core software to add.

I would be interested to know if any existing clients already have this feature added.

**Comment by Cassolotl** [comment]
Duplicate of *Add ALT-Tag to banner and profile pictures and make the value editable for acessiblility reasons #10313* (March 2019)?

**Comment by vmstan** [comment]
Closed as duplicate.
